jetty-server/src/main/java/org/eclipse/jetty/server/HttpConnection.java

```diff
@@ -379,15 +377,15 @@ else if (filled < 0)
     private boolean parseRequestBuffer()
     {
         if (LOG.isDebugEnabled())
-            LOG.debug("{} parse {}", this, BufferUtil.toDetailString(_requestBuffer));
+            LOG.debug("{} parse {}", this, _retainableByteBuffer);
 
-        boolean handle = _parser.parseNext(_requestBuffer == null ? BufferUtil.EMPTY_BUFFER : _requestBuffer);
+        boolean handle = _parser.parseNext(_retainableByteBuffer == null ? BufferUtil.EMPTY_BUFFER : _retainableByteBuffer.getBuffer());
 
         if (LOG.isDebugEnabled())
             LOG.debug("{} parsed {} {}", this, handle, _parser);
 
         // recycle buffer ?
-        if (_contentBufferReferences.get() == 0)
+        if (_retainableByteBuffer != null && _retainableByteBuffer.getReferences() == 1)
```

Comment:
Please add a comment about why we test for `== 1`.

Reply:
I replaced all the `== 1` and `> 1` comparisons with a call to a more readable `boolean isRetained()` method and added some javadoc.